internal/resolver/dns/dns_resolver_test.go

```diff
@@ -736,12 +749,134 @@ func testDNSResolver(t *testing.T) {
 	}
 }
 
+// DNS Resolver immediately starts polling on an error from grpc. This should continue until the ClientConn doesn't
+// send back an error from updating the DNS Resolver's state.
+func TestDNSResolverExponentialBackoff(t *testing.T) {
+	defer leakcheck.Check(t)
+	defer func(nt func(d time.Duration) *time.Timer) {
+		newTimer = nt
+	}(newTimer)
+	timerChan := make(chan *time.Timer, 1)
+	newTimer = func(d time.Duration) *time.Timer {
+		// Will never fire on its own, allows this test to call timer immediately.
+		t := time.NewTimer(time.Hour)
+		timerChan <- t
+		return t
+	}
+	tests := []struct {
+		target   string
+		addrWant []resolver.Address
+		scWant   string
+	}{
+		{
+			"foo.bar.com",
+			[]resolver.Address{{Addr: "1.2.3.4" + colonDefaultPort}, {Addr: "5.6.7.8" + colonDefaultPort}},
+			generateSC("foo.bar.com"),
+		},
+		{
+			"foo.bar.com:1234",
+			[]resolver.Address{{Addr: "1.2.3.4:1234"}, {Addr: "5.6.7.8:1234"}},
+			generateSC("foo.bar.com"),
+		},
+		{
+			"srv.ipv4.single.fake",
+			[]resolver.Address{{Addr: "2.4.6.8" + colonDefaultPort}},
+			generateSC("srv.ipv4.single.fake"),
+		},
+	}
+	for _, a := range tests {
+		b := NewBuilder()
```

Comment:
Would be nicer as:

```go
for _, test := range tests {
	t.Run(test.name, func() {
		// ... test loop body here
	})
}
```

This allows to re-run specific test cases, and prints the test case name (use a short description of the purpose of the case) if something fails.  It also means you can do things like `defer` and they will happen when the test case is done.

Reply:
Changed. Hopefully you like my test names.